src/components/schemas/fields.yaml

```diff
@@ -236,6 +236,149 @@ contains_operator:
     Value signifying the operator for the field
   example: IN
 
+expense_policy_rule_minimal:
+  type: object
+  properties:
+    id:
+      type: string
+      example: tpr1234abcd
+      description: This represents the id of the expense policy rule that is violated for this expense.
+    description:
+      anyOf:
+        - $ref: '#/description'
+      description: |
+        This represents the description of the expense policy rule that is violated for this expense.
+      example: Receipt is mandatory for expense above $50
+    action_show_warning:
+      type: boolean
+      description: |
+        A true value here means that policy calculation suggests showing a warning when the user attempts to create an expense that violates this expense policy rule.
+      example: True
+      
+expense_policy_violating_expenses_query_params:
+  type: object
+  description: |
+    An expense may be violating an expense policy rule due to a group of other expenses. 
+    These query params can be used to fetch those expenses because of which this expense is causing a violation. <br>
+    Below params should be ANDed to get the correct list of violating expenses.
+  properties:
+    start_date:
+      type: string
+      description: |
+        Spend start date query param to find the list of violating expenses.
+      example: "2021-12-01"
+    end_date:
+      type: string
+      description: |
+        Spend end date query param to find the list of violating expenses.
+      example: "2022-03-31"
+    state:
+      type: string
+      description: |
+        State query param to find the list of violating expenses.
+      example: "neq.DRAFT"
+    category_ids:
+      type: string
+      description: |
+        Category ids query param to find the list of violating expenses.
+      example: "in.(60506)"
+    project_ids:
+      type: string
+      description: |
+        Project ids query param to find the list of violating expenses.
+      example: "in.(261155)" 
+    cost_center_ids:
+      type: string
+      description: |
+        Cost center ids query param to find the list of violating expenses.
+      example: "in.(7475)"
+    currency:
+      type: string
+      description: |
+        Currency query param to find the list of violating expenses.
+      example: "USD"
+    vehicle_type:
+      type: string
+      description: |
+        Vehicle type query param to find the list of violating expenses.
+      example: "ELECTRIC_CAR"
+    employee_id:
+      anyOf:
+      - $ref: '#/employee_id'
+      description:
+        Employee id query param to find the list of violating expenses.
+
+expense_policy_individual_desired_states:
+  type: object
+  properties:
+    rule:
+      $ref: '#/expense_policy_rule_minimal'
+    amount:
```

Comment:
{
 policy_rule_id
 amount
 violating_expenses_query_params
}

Reply:
expense_wot (project_id, category_id)
expense_rot (project_id, project {}, category_id, category {})

Reply:
should use a view to mimic rot behaviour